sdk/runtime/runenv.go

```diff
@@ -258,6 +259,41 @@ func (re *RunEnv) BooleanParamD(name string, def bool) bool {
 	return b
 }
 
+// StringArrayParam returns an array of string parameter, or an empty array
+// if it does not exist. The second returns value indicates if the parameter
+// was (un)set or (in)valid.
+func (re *RunEnv) StringArrayParam(name string) (a []string, ok bool) {
+	a = []string{}
+	ok = re.JSONParam(name, &a)
+	return a, ok
+}
+
+// StringArrayParamD returns an array of string parameter or the default one
+// if the parameter was not set or is invalid.
+func (re *RunEnv) StringArrayParamD(name string, def []string) []string {
+	a := []string{}
+	if ok := re.JSONParam(name, &a); !ok {
+		return def
+	}
+	return a
+}
+
+// JSONParam unmarshals a JSON parameter in an arbitrary interface.
+// It returns if the parameter was (un)set or (in)valid.
+func (re *RunEnv) JSONParam(name string, v interface{}) (ok bool) {
+	s, ok := re.TestInstanceParams[name]
+	if !ok {
+		return false
+	}
+
+	if err := json.Unmarshal([]byte(s), v); err != nil {
+		fmt.Println(err)
```

Comment:
I think we shouldn't print errors to stdout, but rather return them to the caller to decide what to do with them, or just ignore them, depending on circumstances.

Reply:
Completely missed that! Thanks for noticing. The plan is to ignore them in this case.